bundler/exe/bundle

```diff
@@ -18,14 +18,17 @@ end
 # Workaround for non-activated bundler spec due to missing https://github.com/rubygems/rubygems/commit/4e306d7bcdee924b8d80ca9db6125aa59ee4e5a3
 gem "bundler", Bundler::VERSION if Gem.rubygems_version < Gem::Version.new("2.6.2")
 
-# Check if an older version of bundler is installed
-$LOAD_PATH.each do |path|
-  next unless path =~ %r{/bundler-0\.(\d+)} && $1.to_i < 9
-  err = String.new
-  err << "Looks like you have a version of bundler that's older than 0.9.\n"
-  err << "Please remove your old versions.\n"
-  err << "An easy way to do this is by running `gem cleanup bundler`."
-  abort(err)
+if Gem::VERSION < "3.2.3" && RUBY_VERSION < "2.6.a" && !ENV["BUNDLER_NO_OLD_RUBYGEMS_WARNING"]
+  Bundler.ui.warn \
+    "Your rubygems version (#{Gem::VERSION})) has a bug that makes dropping " \
+    "support for old rubies in bundler through bumping the " \
+    "`required_ruby_version` field in the bundler gemspec unsafe for any " \
+    "scripts including unconstrained `gem install bundler` commands. That " \
+    "means that, if those scripts are using rubygems #{Gem::VERSION}, they " \
+    "will break as soon as bundler releases a version dropping support for " \
+    "your Ruby version. Please upgrade rubygems to a safe version by running " \
+    "`gem update --system 3.2.3` to avoid future breakage and silence this " \
+    "warning."
```

Comment:
```suggestion
  Bundler.ui.warn \
    "Your RubyGems version (#{Gem::VERSION})) has a bug that prevents " \
    "`required_ruby_version` from working for Bundler. Any scripts that use " \
    "`gem install bundler` will break as soon as Bundler drops support for " \
    "your Ruby version. Please upgrade RubyGems to avoid future breakage " \
    "and silence this warning by running `gem update --system 3.2.3`"
```

Tried to get straight to the problem while keeping your wording. Open to pushback, though. :)

Reply:
I really like it!